src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -675,36 +675,28 @@ private void AddMigratorBar()
             _root.Children.Insert(0, _migratorBar);
         }
 
-#pragma warning disable IDE1006 // Naming Styles
-        private void packageRestoreManager_PackagesMissingStatusChanged(object sender, PackagesMissingStatusEventArgs e)
-#pragma warning restore IDE1006 // Naming Styles
+        private void PackageRestoreManager_PackagesMissingStatusChanged(object sender, PackagesMissingStatusEventArgs e)
         {
-            // make sure update happens on the UI thread.
-            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            // TODO: PackageRestoreManager fires this event even when solution is closed.
+            // Don't do anything if solution is closed.
+            // Add MissingPackageStatus to keep previous packageMissing status to avoid unnecessarily refresh
+            // only when package is missing last time and is not missing this time, we need to refresh
+            if (!e.PackagesMissing && _missingPackageStatus)
             {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                // TODO: PackageRestoreManager fires this event even when solution is closed.
-                // Don't do anything if solution is closed.
-                // Add MissingPackageStatus to keep previous packageMissing status to avoid unnecessarily refresh
-                // only when package is missing last time and is not missing this time, we need to refresh
-                if (!e.PackagesMissing && _missingPackageStatus)
-                {
-                    await UpdateAfterPackagesMissingStatusChangedAsync();
-                }
+                UpdateAfterPackagesMissingStatusChanged();
+            }
 
-                _missingPackageStatus = e.PackagesMissing;
-            });
+            _missingPackageStatus = e.PackagesMissing;
         }
 
         // Refresh the UI after packages are restored.
         // Note that the PackagesMissingStatusChanged event can be fired from a non-UI thread in one case:
         // the VsSolutionManager.Init() method, which is scheduled on the thread pool.
-        private async ValueTask UpdateAfterPackagesMissingStatusChangedAsync()
+        private void UpdateAfterPackagesMissingStatusChanged()
         {
-            VSThreadHelper.ThrowIfNotOnUIThread();
             var timeSinceLastRefresh = GetTimeSinceLastRefreshAndRestart();
-            await RefreshAsync();
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () => await RefreshAsync())
```

Comment:
If you want to start a fire and forget method, you should also put the EmitRefreshEvent and _pacakgeDetail.Refresh with it, otherwise you will not have the refreshed data.

Reply:
Not sure I follow. What refresh data in particular?

the `_packageDetail.Refresh()` itself is already fire and forget, so I thought there's no need to nest it.